server/svix-server/src/v1/endpoints/health.rs

```diff
@@ -1,12 +1,125 @@
 // SPDX-FileCopyrightText: © 2022 Svix Authors
 // SPDX-License-Identifier: MIT
 
-use axum::{http::StatusCode, routing::get, Router};
+use std::time::Duration;
 
-async fn health() -> StatusCode {
+use axum::{http::StatusCode, routing::get, Extension, Json, Router};
+use sea_orm::{query::Statement, ConnectionTrait, DatabaseBackend, DatabaseConnection};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    core::cache::{kv_def, Cache, CacheBehavior, CacheKey, CacheValue},
+    queue::{QueueTask, TaskQueueProducer},
+};
+
+async fn ping() -> StatusCode {
     StatusCode::NO_CONTENT
 }
 
+#[derive(Debug, Deserialize, Serialize)]
+pub enum HealthStatusVariant {
+    Ok,
+    Error,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct HealthStatus {
+    status: HealthStatusVariant,
+    information: String,
+}
+
+impl HealthStatus {
+    pub fn new_ok() -> HealthStatus {
+        HealthStatus {
+            status: HealthStatusVariant::Ok,
+            information: "OK".to_owned(),
+        }
+    }
+
+    pub fn new_error(information: String) -> HealthStatus {
+        HealthStatus {
+            status: HealthStatusVariant::Error,
+            information,
+        }
+    }
+
+    pub fn is_ok(&self) -> bool {
+        matches!(
+            self,
+            HealthStatus {
+                status: HealthStatusVariant::Ok,
+                ..
+            }
+        )
+    }
+}
+impl<O, E: std::error::Error> From<Result<O, E>> for HealthStatus {
+    fn from(res: Result<O, E>) -> Self {
+        match res {
+            Ok(_) => HealthStatus::new_ok(),
+            Err(e) => HealthStatus::new_error(e.to_string()),
```

Comment:
We probably don't want this as it may contain private information (potentially).
So you need to have static strings: "the database timed out", "got an error", "whatever".

Though I think that maybe step one we can just drop information altogether. What do you think?